packages/trail-core/lib/index.js

```diff
@@ -87,16 +87,38 @@ class TrailsManager {
           ("when" >= ${from.toISO()} AND "when" <= ${to.toISO()})
     `
 
+    // Prepare the query to get Count
+    const sqlCount = SQL`
+      SELECT
+          count(*)
+        FROM trails
+        WHERE
+          ("when" >= ${from.toISO()} AND "when" <= ${to.toISO()})
+    `
+
     const op = caseInsensitive ? 'ILIKE' : 'LIKE'
-    if (who) sql.append(SQL([` AND who_id ${op} `])).append(SQL`${exactMatch ? who : '%' + who + '%'}`)
-    if (what) sql.append(SQL([` AND what_id ${op} `])).append(SQL`${exactMatch ? what : '%' + what + '%'}`)
-    if (subject) sql.append(SQL([` AND subject_id ${op} `])).append(SQL`${exactMatch ? subject : '%' + subject + '%'}`)
+    if (who) {
+      sql.append(SQL([` AND who_id ${op} `])).append(SQL`${exactMatch ? who : '%' + who + '%'}`)
+      sqlCount.append(SQL([` AND who_id ${op} `])).append(SQL`${exactMatch ? who : '%' + who + '%'}`)
+    }
+    if (what) {
+      sql.append(SQL([` AND what_id ${op} `])).append(SQL`${exactMatch ? what : '%' + what + '%'}`)
+      sqlCount.append(SQL([` AND what_id ${op} `])).append(SQL`${exactMatch ? what : '%' + what + '%'}`)
+    }
+    if (subject) {
+      sql.append(SQL([` AND subject_id ${op} `])).append(SQL`${exactMatch ? subject : '%' + subject + '%'}`)
+      sqlCount.append(SQL([` AND subject_id ${op} `])).append(SQL`${exactMatch ? subject : '%' + subject + '%'}`)
+    }
 
     const footer = ` ORDER BY ${sortKey} ${sortAsc ? 'ASC' : 'DESC'} LIMIT ${pageSize} OFFSET ${(page - 1) * pageSize}`
     sql.append(SQL([footer]))
 
     const res = await this.performDatabaseOperations(sql)
-    return res.rows.map(convertToTrail)
+
+    // Get record count if the flag is set to true
+    const resCount = await this.performDatabaseOperations(sqlCount)
+
+    return { count: resCount.rows[0].count, data: res.rows.map(convertToTrail) }
```

Comment:
the two queries must be run within the same transaction. see how performDatabaseOperations is implemented, it should support what you're trying to do already

Reply:
did you mean like this 
```
const result = await this.performDatabaseOperations( async client => {
      const count = await client.query(sqlCount);
      const results = await client.query(sql)
      return { count: parseInt(count.rows[0].count), data: results.rows.map(convertToTrail)}
    })
```

Reply:
However with this change, the following test breaks since it is not a string parameter any more
`test('should sanitize pagination parameters' `

Is it ok to remove this test ?

Reply:
can you please link the test? I'm not sure I understand how an existing test would be affected by a new property you added that didn't exist before

Reply:
I am referring to this test. The `performDatabaseOperations` parameter is now changed to a function instead of the string earlier.
https://github.com/nearform/trail/blob/b8387dcf7b73b276d12090f7b1cfe9ef52427c8f/packages/trail-core/test/index.test.js#L199-L218